esx_service/utils/auth_api.py

```diff
@@ -22,42 +22,68 @@
 import error_code
 import log_config
 import logging
+from error_code import ErrorCode
+from error_code import ErrorInfo
+
+def generate_error_info_for_internal_error(secondary_err_msg):
+    err_code = ErrorCode.INTERNAL_ERROR
+    err_msg = error_code.error_code_to_message[err_code].format(secondary_err_msg)
+    error_info = ErrorInfo(err_code, err_msg)
+    return error_info
+
+def generate_error_info_for_tenant_not_exist(name):
+     err_code = ErrorCode.TENANT_NOT_EXIST
+     err_msg = error_code.error_code_to_message[err_code].format(name)
+     error_info = ErrorInfo(err_code, err_msg)
+     return error_info  
+
+def generate_error_info_for_vm_not_found(not_found_vms):
+    not_found_vm_list = ",".join(not_found_vms)
+    err_code = ErrorCode.VM_NOT_FOUND
+    err_msg = error_code.error_code_to_message[err_code].format(not_found_vm_list)
+    error_info = ErrorInfo(err_code, err_msg)
+    return error_info
+
```

Comment:
Nit - since these functions are generating the error info could rename so that the action is implicit like vm_not_found(), internal_error(), etc.